File: 0374-guess-number-higher-or-lower/0374-guess-number-higher-or-lower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if num is higher than the picked number
#          1 if num is lower than the picked number
#          otherwise return 0
# def guess(num: int) -> int:

class Solution:
    def guessNumber(self, n: int) -> int:
        lower_limit = 1
        upper_limit = n+1
        my_guess = 0
        while True:
            my_guess = (upper_limit+lower_limit)//2
            result = guess(my_guess)
            if result == -1:
                upper_limit = my_guess
            elif result == 1:
                lower_limit = my_guess
            else:
                break
        
        return my_guess

File: 0374-guess-number-higher-or-lower/0374-guess-number-higher-or-lower.cpp
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * int guess(int num);
 */

class Solution {
public:
    int guessNumber(int n) {
        int lower, my_guess, result;
        unsigned int upper;
        lower = 1;
        upper = n;
        upper += 1;
        while(true){
            my_guess = (upper+lower)/2;
            result = guess(my_guess);
            if(result == 0){
                return my_guess;
            }
            else if(result == 1){
                lower = my_guess;
            }
            else{
                upper = my_guess;
            }
        }
    }
};